src/volume_calc.cpp

```diff
@@ -99,9 +99,9 @@ vector<VolumeCalculation::Result> VolumeCalculation::execute() const
 {
   // Shared data that is collected from all threads
   int n = domain_ids_.size();
-  vector<vector<int>> master_indices(
+  vector<vector<int32_t>> master_indices(
     n); // List of material indices for each domain
-  vector<vector<int>> master_hits(
+  vector<vector<size_t>> master_hits(
```

Comment:
You'll need associated changes in the MPI communication. The indices/hits get packed into a single buffer, so you'll need them to be the same type. Also, there's no corresponding MPI datatype for `size_t`, so you might be better off using `uint64_t`, for which the MPI type is `MPI_UINT64_T`.

Reply:
Ah, yeah you're totally right, I didn't consider this. I'll take a closer look and switch over to `uint64_t`.